Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/ModifyProjectController.java

```diff
@@ -39,230 +37,177 @@
+        ModelAndView model;
+        String userName = principal.getName();
+        if (result.hasErrors()) {
+            model = new ModelAndView("auth/workbench/modifyproject");
+            model.getModelMap().put("project", project);
+            return model;
+        }
+        projectManager.updateProject(project.getProjectId(), project.getProjectName(), project.getDescription(),
+                project.getProjectAccess().name(), userName);
+        redirectAttributes.addFlashAttribute("show_success_alert", true);
+        redirectAttributes.addFlashAttribute("success_alert_msg", "Project has been updated successfully.");
+        model = new ModelAndView("redirect:/auth/workbench/projects/" + projectid);
+        return model;
+    }
+
+    /**
+     * This controller function would assign editor roles to project owner
+     * 
+     * @param projectId
+     * @param model
+     * @param principal
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     */
+    @RequestMapping(value = "auth/workbench/assignownereditor/{projectid}", method = RequestMethod.GET)
+    public String assignOwnerEditorRole(@PathVariable("projectid") String projectId, ModelMap model,
+            Principal principal) throws QuadrigaStorageException, QuadrigaAccessException {
+        IUser user = userManager.getUser(principal.getName());
+        String userName = user.getUserName();
+        projectManager.assignEditorRole(projectId, userName);
+        IProject project = retrieveProjectManager.getProjectDetails(projectId);
+
+        // retrieve all the workspaces associated with the project
+        List<IWorkSpace> workspaceList = wsManager.listActiveWorkspace(projectId, userName);
+        if (projectSecurity.isProjectOwner(userName, projectId)) {
+            model.addAttribute("owner", 1);
+        } else {
+            model.addAttribute("owner", 0);
+        }
+        if (projectSecurity.isEditor(userName, projectId)) {
+            model.addAttribute("editoraccess", 1);
+        } else {
+            model.addAttribute("editoraccess", 0);
+        }
+        model.addAttribute("project", project);
+        model.addAttribute("workspaceList", workspaceList);
+        return "auth/workbench/project";
+    }
+
+    /**
+     * This controller function would assign editor roles to project owner
+     * 
+     * @param projectId
+     * @param model
+     * @param principal
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     */
+    @RequestMapping(value = "auth/workbench/deleteownereditor/{projectid}", method = RequestMethod.GET)
+    public String deleteOwnerEditorRole(@PathVariable("projectid") String projectId, ModelMap model,
+            Principal principal) throws QuadrigaStorageException, QuadrigaAccessException {
+        IUser user = userManager.getUser(principal.getName());
+        String userName = user.getUserName();

```

Comment:
this is unnecessary I believe; you can use principal.getName to get the username